### PR TITLE
Release 2.0.0

### PR DESCRIPTION
**Pull request by danielbitzer**
Releases a new version of the framework with a tiny change to help https://github.com/woocommerce/facebook-for-woocommerce/pull/1924

- This PR also adds a changelog. I chose the Gutenberg format so we can have issue links and formatting.
- I also added a note about breaking changes
- See released PR #2 

Thoughts and feedback welcome. I'm somewhat rushing this out so we can keep FB feed work moving.

### CODE REVIEW

**Comment by danbitzer** [comment]
Also @haszari is it possible to add @woocommerce/tetris to this repo. I don't have permission to do that.

**Review by budzanowski — APPROVED** [review]

**Comment by budzanowski** [comment]
:shipit:

File: composer.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "name": "woocommerce/action-scheduler-job-framework",
     "description": "A job framework for Action Scheduler (actionscheduler.org)",
-    "version": "1.0.0",
+    "version": "1.1.0",
```
Comment (haszari) [inline composer.json]: Semantic? WordPress versioning? Might as well decide on a convention now :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<!-- For now we follow Gutenberg's package changelog format https://github.com/WordPress/gutenberg/tree/HEAD/packages#maintaining-changelogs. -->
+
+## 1.1.0 (2021-05-19)
+
+### Breaking Changes
+
+- Removed `...AbstractChainedJob::filter_items_before_processing` method ([#2](https://github.com/woocommerce/action-scheduler-job-framework/pull/2))
+
+### New Feature
+- Added `...AbstractChainedJob::process_items` method ([#2](https://github.com/woocommerce/action-scheduler-job-framework/pull/2))
```
Comment (haszari) [inline CHANGELOG.md]: Nice to use markdown format! Issue/PR links help clarify changes, can dig back through to get the context or history.

With these changes specifically, the changelog items could focus more on why - what the change enables the library client to do.

**Review by haszari — APPROVED** [review]
Awesome to have some clarity around changes and releases/versioning here. I've added some light feedback about various things but nothing blocking. 

The one I care about most is making changelog items focus on why (rather than what/how), but that's not a blocker. 🚢

**Comment by haszari** [comment]
> Also @haszari is it possible to add @woocommerce/tetris to this repo. I don't have permission to do that.

Me neither @danielbitzer !

@jconroy  can you add more of us to this repo? I'm thinking all of Ventures, as this repo may be of use in other product sync/batch processing code.

File: CHANGELOG.md (hunk 2 above)
Comment (haszari) [reply CHANGELOG.md]: I took a stab after seeing the change in context on https://github.com/woocommerce/facebook-for-woocommerce/pull/1924#discussion_r634974904.

> Added `AbstractChainedJob::process_items()` method. Clients can override to use state across the batch job.

(I'm finding this tricky to explain!)

File: composer.json (hunk 1 above)
Comment (danbitzer) [reply composer.json]: I feel like for "not-plugins" projects semantic versioning is the norm. Are you suggesting it should be 2.0.0 to be semantic?

File: CHANGELOG.md (hunk 2 above)
Comment (danbitzer) [reply CHANGELOG.md]: Sounds good, I added some more context in 48c94e2e01ebe9acff8be687c1aecfb8883d98af

File: composer.json (hunk 1 above)
Comment (haszari) [reply composer.json]: Not making any suggestion - happy with whatever you decide. I agree, 1.1 is semantic and signals the breaking change, and semantic works for this project IMO :)

So next release would be 1.1.1 right - unless it had breaking changes.

Comment (danbitzer) [reply composer.json]: Technically it should be 2.0 due to the breaking change. Minor versions are supposed to be fully backwards compatible...

https://semver.org/

Comment (haszari) [reply composer.json]: Interesting! Well my only feedback is be explicit about what the versioning policy is.

Comment (danbitzer) [reply composer.json]: Sounds good, I've updated the readme to state it uses Semver and bumped this version to 2.0.0.

Comment (haszari) [reply composer.json]: 💥

**Pull request merged**